scripts/cnv_wdl/cnv_common_tasks.wdl

```diff
@@ -605,3 +612,100 @@ task CollectModelQualityMetrics {
         String qc_status_string = read_string("qcStatus.txt")
     }
 }
+
+task TransposeCallerOutputs {
+    input {
+      Array[File] gcnv_calls_tars
+
+      # Runtime parameters
+      String docker
+      Int? mem_gb
+      Int? disk_space_gb
+      Boolean use_ssd = false
+      Int? cpu
+      Int? preemptible_attempts
+    }
+
+    command <<<
+        set -euo pipefail
+
+        gcnv_calls_tar_array=(~{sep=" " gcnv_calls_tars})
+        for index in ${!gcnv_calls_tar_array[@]}; do
+            mkdir CALLS_$index
+            tar xzf ${gcnv_calls_tar_array[$index]} -C CALLS_$index
+        done
+
+        CURRENT_SAMPLE=0
+        NUM_SAMPLES=$(ls -d CALLS_0/SAMPLE_* | wc -l)
+        NUM_DIGITS=${#NUM_SAMPLES}
+        while [ $CURRENT_SAMPLE -lt $NUM_SAMPLES ]; do
+            CURRENT_SAMPLE_WITH_LEADING_ZEROS=$(printf "%0${NUM_DIGITS}d" $CURRENT_SAMPLE)
+            tar c CALLS_*/SAMPLE_$CURRENT_SAMPLE | gzip -1 > gcnv-calls-sample-$CURRENT_SAMPLE_WITH_LEADING_ZEROS.tar.gz
+            let CURRENT_SAMPLE=CURRENT_SAMPLE+1
+        done
+
+        rm -r CALLS_*/SAMPLE_*
+        tar c CALLS_* | gzip -1 > gcnv-shard-configs.tar.gz
+    >>>
+
+    runtime {
+        docker: docker
+        memory: select_first([mem_gb, 2]) + " GiB"
+        disks: "local-disk " + select_first([disk_space_gb, 150]) + if use_ssd then " SSD" else " HDD"
```

Comment:
```suggestion
    Int disk_baseline_gb_ = 10
    Float compression_factor = 10.0
    Int disk_needed_gb = disk_baseline_gb + ceil(compression_factor * size(gcnv_calls_tars, "GiB"))
    
    runtime {
        docker: docker
        memory: select_first([mem_gb, 2]) + " GiB"
        disks: "local-disk " + select_first([disk_space_gb, disk_needed_gb]) + if use_ssd then " SSD" else " HDD"
```
I don't see memory being a problem here, but these kinds of tasks have been running of out disk on larger panels. This should be conservative enough without being super wasteful.

Reply:
Thanks. This would have been good but it turns out we aren't using this task. It might be useful to have automatic scaling for the rest of the gCNV tasks but I'll leave this for later work.